service/collector/trusted_advisor.go

```diff
@@ -237,17 +358,13 @@ func (t *TrustedAdvisor) getTrustedAdvisorResources(id *string, awsClients aws.C
 	return checkResultOutput.Result.FlaggedResources, nil
 }
 
-func resourceToMetrics(resource *support.TrustedAdvisorResourceDetail, accountID string) (prometheus.Metric, prometheus.Metric, error) {
-	if len(resource.Metadata) != resourceMetadataLength {
-		return nil, nil, invalidResourceError
-	}
-
-	region := resource.Metadata[indexRegion]
-	service := resource.Metadata[indexService]
-	limitName := resource.Metadata[indexName]
+func resourceToMetrics(resource trustedAdvisorResource, accountID string) (prometheus.Metric, prometheus.Metric, error) {
+	region := resource.Region
+	service := resource.Service
+	limitName := resource.LimitName
 
-	limit := resource.Metadata[indexLimit]
-	usage := resource.Metadata[indexUsage]
+	limit := resource.Limit
```

Comment:
WDYT, can we use the resource fields directly instead of assigning them again?

Reply:
True, we probably could just cache the `resources` we get from the AWS api without having another struct. I'll try that.